### PR TITLE
deps: Bump golang (PROJQUAY-4556)

### PR DESCRIPTION
**Pull request by HammerMeetNail**
* Updates golang versin to `1.17.12`

### CODE REVIEW

**Comment by HammerMeetNail** [comment]
Need to verify if this breaks FIPS...

**Comment by HammerMeetNail** [comment]
Hmm, might already be broken. Can't spin up v3.7.8 or v3.8.0 in a FIPS cluster.

File: Dockerfile [hunk 1]
```diff
@@ -103,13 +103,13 @@ RUN set -ex\
...
 # Config-tool builds the go binary in the configtool.
-FROM registry.access.redhat.com/ubi8/go-toolset:1.16.12 as config-tool
+FROM registry.access.redhat.com/ubi8/go-toolset:1.17.12 as config-tool
 WORKDIR /opt/app-root/src
 ARG CONFIGTOOL_VERSION=v0.1.12
 RUN curl -fsSL "https://github.com/quay/config-tool/archive/${CONFIGTOOL_VERSION}.tar.gz"\
 	| tar xz --strip-components=1 --exclude '*/pkg/lib/editor/static/build'
 COPY --from=config-editor /opt/app-root/src/static/build  /opt/app-root/src/pkg/lib/editor/static/build
-RUN go install ./cmd/config-tool
+RUN go install -tags=fips ./cmd/config-tool
```
Comment (jonathankingfc) [inline Dockerfile:112]: Where does this tag come from?

Comment (HammerMeetNail) [reply Dockerfile:112]: See https://github.com/minio/minio-go/pull/1700 and this comment https://github.com/minio/minio-go/issues/1697#issuecomment-1259523447

Comment (jonathankingfc) [reply Dockerfile:112]: LGTM

**Review by jonathankingfc — APPROVED** [review]

**Pull request merged**